### ISSUE TITLE
Worker get more than 1 job

### ISSUE THREAD

**Issue by allancarvalho** [scttnlsn/monq#58]
Hello

I'm using monq to process a queue with 8000 items +-
The jobs are: call a service, get a response and upload static file with a response in FTP.

I need to do the same job every 10 minutes because I need to update the static files.

Today I'm running monq with docker, and for speed up the queues I need to scale the application.

But each container always get only 1 job and I think that could get more

I did this already, but before to send the pull request I would like to have more opinions about this.

**Comment by scttnlsn** [comment]
This sounds like a duplicate of https://github.com/scttnlsn/monq/pull/52

**Comment by allancarvalho** [comment]
No! It's not the same problem.

For exemple.

I have 5 containers.
Each containers get 1 job.
So, I have 5 jobs being processed.

My suggestion is: each container get more then 1 job parallel.

**Comment by allancarvalho** [comment]
I want to do more tests before send pull request, but the idea is it

https://github.com/allancarvalho/monq/commit/22b34fcea7c607f1a76a5af5222162d668004844

**Comment by scttnlsn** [comment]
Why not just start multiple workers in the same process?

**Comment by allancarvalho** [comment]
I think that not understood.

I can register multiple workers for multiple queues, but not multiples workers for 1 queue.
Today, for do it, I need to scale the application. It is good, but I think that could do more if each worker get more jobs.

**Comment by scttnlsn** [comment]
Could you be more specific about what is preventing you from starting multiple workers (that read from the same queue) inside of a single process?

**Comment by Santinell** [comment]
How that must be - all queues processed in parallel, all jobs in queue processed in series.
If we do as you say, what will be:
Each worker will process one job in one time from all queues in series.
That means, if I had two queues: `One` and `Two`, which have 1 and 6 jobs accordingly,
after first job from queue `One` is processed, all remaining jobs from queue `Two` will be processed in parallel, but must be in series.

**Comment by scttnlsn** [comment]
If you need jobs on a particular queue to be processed serially then there can only be a single worker reading off of that queue.  If jobs can be processed in parallel then feel free to start multiple workers reading from the same queue (in the same or separate processes).

**Comment by scttnlsn** [comment]
Closing due to inactivity.  Please re-open if you'd like to continue discussing this.
